### DISCUSSION TITLE
Implement Legacy Error Hit Bar as a Skin Layout Element

### DISCUSSION THREAD

**Discussion by Methos-** [ppy/osu#17236] [Ideas] [answered]
Not quite sure why we don't have this since we have everything else from the Legacy client. It's been a reason why myself and a few others I have spoken to have been opposed to switching over to Lazer as their new client (ignoring scoring).

The error bar is very easy to read just due to color contrast and hitmarkers cutting all the way through the graph, blue marks will highlight to nearly white when hit accurately in succession. You don't quite get this with the new UR bar.

**Comment by TheFantasticLoki** [comment]
Legacy Input overlay is also missing so maybe that can be implemented soon.

**Reply by frenzibyte** [reply]
Open a separate discussion thread for that. We can't track including two different features in one issue/discussion thread.

**Comment by frenzibyte** [comment]
I believe the "legacy hit error meter" isn't implemented because it isn't skinnable in the first place, and if the current one isn't easily readable, then that warrants improving it rather than bringing old designs back.

For visual reference, this is how both currently look:
| stable | lazer |
|--------|-------|
| ![CleanShot 2022-03-14 at 02 39 56](https://user-images.githubusercontent.com/22781491/158084474-9f118680-ddac-4d7f-b604-84d3b944f92e.gif) | ![CleanShot 2022-03-14 at 02 42 10](https://user-images.githubusercontent.com/22781491/158084485-3c41b21f-0de0-4ce1-a3d4-609f25aa5644.gif) |

The hits on the lazer variant do seem a bit unreadable, unlike stable. Might want them extend over the meter rather than appearing as ticks on the top to be readable.

**Reply by peppy** [reply]
@arflyte do you have any updated/better designs for this? the lazer one is indeed pretty hard to read but this was done per the provided design. shall we start by making the ticks taller?

**Comment by arflyte** [comment]
Yeah, I have to agree lazer's design is not effective. let's make the ticks bigger for now. I'll keep this in mind for the latest design.

**Comment by peppy** [comment, marked as answer]
Have improved things a bit over at #17245. Can offer more customisation once the skin settings change is live, too.

**Reply by PercyDan54** [reply]
For me the circle as a center indicator looks weird

**Reply by Methos-** [reply]
A few people so far have told me the hit indicator ticks are a bit thick as well.

Might want see if there is a balance you can find with a small testing group.

**Reply by peppy** [reply]
There will be further customisation available for those who want to change the appearance. I'm happy with the current settings as a default for now.

**Comment by San-Mei** [comment]
This is now merged: https://github.com/ppy/osu/pull/38535
